### PR TITLE
Implement rule avoid-returning-widgets

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[X] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

**Comment by GroovinChip** [comment]
VERY happy to see this 🥳

File: lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import '../../models/severity.dart';
+import '../../utils/node_utils.dart';
+import '../../utils/rule_utils.dart';
+import 'obsolete_rule.dart';
+
+class AvoidReturningWidgets extends ObsoleteRule {
+  static const String ruleId = 'avoid-returning-widgets';
+  static const _documentationUrl = '';
+
+  static const _warningMessage = 'Avoid returning widgets from a function';
```
Comment (dkrutskikh) [inline lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart:17]: @incendial I think a good idea use the rules of [parallel construction](https://en.wikipedia.org/wiki/Parallelism_(grammar))

File: lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                source: source,
+                withCommentOrMetadata: true,
+              ),
+              message: _warningMessage,
+            ))
+        .toList(growable: false);
+  }
+
+  static List<String> _getIgnoredNames(Map<String, Object> config) =>
+      config.containsKey('ignored-names') && config['ignored-names'] is Iterable
```
Comment (dkrutskikh) [inline lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart:49]: @incendial please extract to constant

File: lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import '../../utils/rule_utils.dart';
+import 'obsolete_rule.dart';
+
+class AvoidReturningWidgets extends ObsoleteRule {
+  static const String ruleId = 'avoid-returning-widgets';
+  static const _documentationUrl = '';
+
+  static const _warningMessage = 'Avoid returning widgets from a function';
+
+  final List<String> _ignoredNames;
```
Comment (dkrutskikh) [inline lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart:19]: @incendial please use Iterable

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart (hunk 3 above)
Comment (incendial) [reply lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart]: Done.

File: lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart (hunk 2 above)
Comment (incendial) [reply lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart]: Done.

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**